File: langs/ltsv.rb
def ltsv(line)
  ret = {}
  line.split("\t").each { |token|
    k, v = token.split(":", 2)
    ret[k] = v
  }
  ret
end

record = nil
$stdin.each_line(chomp: true) do |line|
  record = ltsv(line)
end

puts(record)

File: langs/ltsv.py
import sys

def ltsv(line):
    ret = {}
    line = line.strip()
    for token in line.split("\t"):
        k, v = token.split(":", maxsplit=1)
        ret[k] = v
    return ret



for line in sys.stdin:
    record = ltsv(line)


print(record)
